bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/api/IAqualinkClient.java

```diff
@@ -0,0 +1,448 @@
+                String.format(IAQUALINK_URL + "&temp1=%s", "set_temps", serialNumber, sessionId, spaSetpoint),
+                Home.class);
+    }
+
+    /**
+     * Sets the Pool Temperature Setpoint
+     *
+     * @param serialNumber
+     * @param sessionId
+     * @param poolSetpoint
+     * @return
+     * @throws IOException
+     * @throws NotAuthorizedException
+     */
+    public Home setPoolTemp(@Nullable String serialNumber, @Nullable String sessionId, int poolSetpoint)
+            throws InterruptedException, IOException, NotAuthorizedException {
+        return getAqualinkObject(
+                String.format(IAQUALINK_URL + "&temp2=%s", "set_temps", serialNumber, sessionId, poolSetpoint),
+                Home.class);
+    }
+
+    /**
+     * Sends a OneTouch command
+     *
+     * @param serialNumber
+     * @param sessionId
+     * @param command
+     * @return
+     * @throws IOException
+     * @throws NotAuthorizedException
+     */
+    public OneTouch[] oneTouchCommand(@Nullable String serialNumber, @Nullable String sessionId, String command)
+            throws InterruptedException, IOException, NotAuthorizedException {
+        return getAqualinkObject(String.format(IAQUALINK_URL, command, serialNumber, sessionId), OneTouch[].class);
+    }
+
+    /**
+     *
+     * @param <T>
+     * @param url
+     * @param typeOfT
+     * @return
+     * @throws IOException
+     * @throws NotAuthorizedException
+     */
+    private <T> T getAqualinkObject(@Nullable String url, Type typeOfT)
+            throws InterruptedException, IOException, NotAuthorizedException {
+        try {
+            logger.trace("Trying {}", url);
+            ContentResponse response = httpClient.newRequest(url).method(HttpMethod.GET)//
+                    .agent(HEADER_AGENT) //
+                    .header(HttpHeader.ACCEPT_LANGUAGE, HEADER_ACCEPT_LANGUAGE) //
+                    .header(HttpHeader.ACCEPT_ENCODING, HEADER_ACCEPT_ENCODING) //
+                    .header(HttpHeader.ACCEPT, HEADER_ACCEPT) //
+                    .send();
+
+            logger.trace("Response {}", response.toString());
+            if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                throw new NotAuthorizedException(response.getReason());
+            }
+            if (response.getStatus() != HttpStatus.OK_200) {
+                throw new IOException(response.getReason());
+            }
+            return gson.fromJson(response.getContentAsString(), typeOfT);
+        } catch (TimeoutException | ExecutionException | JsonParseException e) {
+            throw new IOException(e);
+        }
+    }
+
+    /**
+     *
+     * @param url
+     * @return
+     * @throws IOException
+     * @throws NotAuthorizedException
+     */
+    private JsonElement getAqualinkJsonObject(String url)
+            throws InterruptedException, IOException, NotAuthorizedException {
+        try {
+            logger.trace("Trying {}", url);
```

Comment:
This looks like it would be better of to share some code with the method above

Reply:
Agreed

Reply:
Agreed